src/nfa/operations.cc

```diff
@@ -1207,27 +1205,68 @@ OrdVector<Symbol> mata::nfa::get_symbols_to_work_with(const Nfa& nfa, const mata
 
 std::optional<mata::Word> Nfa::get_word(const Symbol first_epsilon) const {
     if (initial.empty() || final.empty()) { return std::nullopt; }
+    if (initial.intersects_with(final)) { return Word{}; }
 
-    std::vector<std::pair<State, Word>> worklist{};
+    /// Current state state post iterator, its end iterator, and iterator in the current symbol post to target states.
+    std::vector<std::tuple<StatePost::const_iterator, StatePost::const_iterator, StateSet::const_iterator>> worklist{};
+    std::vector<bool> searched(num_of_states(), false);
+    bool final_found{};
     for (const State initial_state: initial) {
         if (final.contains(initial_state)) { return Word{}; }
-        worklist.emplace_back(initial_state, Word{});
-    }
-    std::vector<bool> searched(num_of_states());
+        if (searched[initial_state]) { continue; }
```

Comment:
This seems unnecessary, you iterate trought initial states once and set all of them to true, so at this point `searched[initial_state]` must be `false`.

Reply:
Ok, maybe it is necessary

Reply:
Nope, it should be unnecessary :D